### DISCUSSION TITLE
Building librdkafka from source in CentOS 7 with libssl 1.1.1

### DISCUSSION THREAD

**Discussion by ganibc** [edenhill/librdkafka#4170] [Q&A] [answered]
Hi,

My build and production machine is using centOS 7 and CentOS7 is using openssl 1.0.2.
If I build librdkafka from source using `./configure`, it will automatically use openssl 1.0.2.
Is there any flag in configure so I can force the build points to the openssl11 directory?
I've tried using LIBS, includedirs, libdirs but it doesn't work. The librdkafka.so still use openssl 1.0.2 that installed in the machine instead of from openssl11 directory.

I read https://github.com/confluentinc/librdkafka/issues/2740, but this doesn't give working solution.

This is what I tried.
First, without specify openssl directory:
```
./configure --source-deps-only --enable-static \
    --disable-gssapi --disable-zlib --disable-zstd \
    --disable-sasl --disable-curl --disable-lz4-ext --disable-regex-ext --disable-c11threads
```
ldd the libssl.so
```
ldd src/librdkafka.so
<... removed the rest>
        libssl.so.10 => /lib64/libssl.so.10 (0x00007fee00967000)
        libcrypto.so.10 => /lib64/libcrypto.so.10 (0x00007fee00504000)
```
You can see the ldd showing libssl 1.0 is linked.

install openssl11 and specify openssl11 dir. I installed using `yum install openssl11-devel`
```
./configure --source-deps-only --enable-static \
    --disable-gssapi --disable-zlib --disable-zstd \
    --disable-sasl --disable-curl --disable-lz4-ext --disable-regex-ext --disable-c11threads \
    --libdir=/usr/local/openssl11/lib --includedir=/usr/local/openssl11/include
```
ldd doesn't show libssl or libcrypto anymore

**Comment by ganibc** [comment, marked as answer]
This has been resolved by removing the --enable-static.

**Comment by Mrsvkarthik** [comment]
Prefix your SSL 1.1 lib paths in LD_LIBRARY_PATH 
setenv LD_LIBRARY_PATH <SSL 1.1 lib paths>:<already existing path> and try building
